kong/plugins/key-auth/handler.lua

```diff
@@ -149,6 +150,11 @@ end
 function KeyAuthHandler:access(conf)
   KeyAuthHandler.super.access(self)
 
+  -- check if preflight request and wether it should be authenticated
+  if conf.authenticate_preflight and get_method() == "OPTIONS" then
```

Comment:
According to this condition, it seems that `authenticate_preflight = true` means "**ignore** and proxy upstream".

However, one can interpret `authenticate_preflight` as to mean "**enforce** the key verification (authentication step) even if preflight". Which is the opposite, and which would mean that `authenticate_preflight = true` would mean: **enforce** instead of **ignore**.

I think the name is thus confusing, and does not carry enough meaning to infer its functionality. In the CORS plugin, we have `preflight_continue` which, if enabled, will make the plugin **ignore**, and thus proxy the request upstream. I thus see 2 possibilities:

1. use the same terminology as the CORS plugin: `config.preflight_continue = true` means: **ignore this plugin on preflight, got directly upstream)
2. Rename to `config.run_on_preflight`, which is a more explicit name

1. is consistent with the CORS, and 2. carries (I feel) the most meaning.

**Note**: this is yet another instance of plugins deciding for themselves when to run and when not to. Something they shouldn't be allowed to do in the future :)